code/controllers/subsystem/minimaps.dm

```diff
@@ -46,15 +46,19 @@ SUBSYSTEM_DEF(minimaps)
 				var/turf/location = locate(xval,yval,level)
 				if(isspaceturf(location))
 					continue
-				if(location.density)
+				else if(location.density)
 					icon_gen.DrawBox(location.minimap_color, xval, yval)
 					continue
-				var/atom/movable/alttarget = locate(/obj/machinery/door) in location||locate(/obj/structure/fence) in location
+				var/atom/movable/alttarget = (locate(/obj/machinery/door) in location) || (locate(/obj/structure/fence) in location)
 				if(alttarget)
 					icon_gen.DrawBox(alttarget.minimap_color, xval, yval)
 					continue
 				var/area/turfloc = location.loc
-				icon_gen.DrawBox(turfloc.minimap_color, xval, yval)
+				if(turfloc.minimap_color)
+					icon_gen.DrawBox(BlendRGB(location.minimap_color, turfloc.minimap_color, 0.5), xval, yval)
```

Comment:
whys there a blendrbg call here and why does it if

Reply:
So with my changes, turf colour always applies, then if its area has a colour (defaulted to null) it will blend the area and turf colour. So if you look at the magmoor picture as an example, you can still see details like lava when they're underground, as it is blending the lava and underground area colour.